### DISCUSSION TITLE
supported types for custom command parameters

### DISCUSSION THREAD

**Discussion by mvolkmann** [nushell/nushell#3175] [Q&A] [answered]
Based on trial and error it seems that the supported types that can be specified on custom command parameters are
`int`, `number` (for float), `path` (for file paths), `pattern` (for glob patterns?), `range`, and `string`.
Is this correct or are there others that are currently supported?

**Comment by fdncred** [comment, marked as answer]
I believe [this](https://github.com/nushell/nushell/blob/56adc7c3c6c8c011c2db9cdcf780d9213a925a8b/crates/nu-parser/src/parse/def/primitives.rs#L162-L185) is where the currentt list of data types for custom commands are defined.

**Comment by e2dk4r** [comment]
If there is someone trying to find answer for nushell `0.101.0` [here](https://github.com/nushell/nushell/blob/fb2610904997f54f50ad52095948bede81695f31/crates/nu-parser/src/parse_shape_specs.rs#L39-L72)
